src/containers/pages/IRS/performanceReport/reports/index.tsx

```diff
@@ -0,0 +1,375 @@
+  breadCrumbs: Page[];
+  cellComponent: React.ElementType;
+  columns: Array<DrillDownColumn<Dictionary<{}>>>;
+  currentPage: Page;
+  dataCollectorsData: IRSCollectorPerformance[];
+  districtsData: IRSDistrictPerformance[];
+  fetchDataCollectors: typeof FetchIRSDataCollectors;
+  fetchDistricts: typeof FetchIRSDistricts;
+  fetchSOPs: typeof FetchIRSSOPs;
+  fetchSopByDate: typeof FetchIRSSOPByDate;
+  linkerField: string;
+  pageTitle: string;
+  service: typeof supersetFetch;
+  sopData: IRSSOPPerformance[];
+  tableData: Array<Dictionary<any>>;
+  urlParamField: string | null;
+}
+
+/** IRSPerfomenceReport displays districts, data collectors, sop and evevent dates IRS performance reports  */
+const IRSPerfomenceReport = (
+  props: IRSPerfomenceReportProps & RouteComponentProps<RouteParams>
+) => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const { params } = props.match;
+  const { planId, jurisdictionId, dataCollector, sop } = params;
+
+  const {
+    pageTitle,
+    service,
+    columns,
+    fetchDataCollectors,
+    fetchDistricts,
+    fetchSOPs,
+    fetchSopByDate,
+    tableData,
+    breadCrumbs,
+    currentPage,
+    cellComponent,
+    urlParamField,
+    linkerField,
+    dataCollectorsData,
+    districtsData,
+    sopData,
+  } = props;
+
+  const breadcrumbProps = {
+    currentPage,
+    pages: [
+      {
+        label: HOME,
+        url: HOME_URL,
+      },
+      ...breadCrumbs,
+    ],
+  };
+
+  /** async function to load the data */
+  async function loadData() {
+    setLoading(tableData.length < 1);
+    try {
+      if (planId && !districtsData.length) {
+        const fetchPlansParams = supersetFilters(params, planId);
+        await service(SUPERSET_IRS_DISTRICT_PERFORMANCE_REPORT_SLICE, fetchPlansParams).then(
+          result => {
+            fetchDistricts(result);
+          }
+        );
+      }
+      if (planId && jurisdictionId && !dataCollectorsData.length) {
+        const fetchPlansParams = supersetFilters(params, jurisdictionId);
+        await service(SUPERSET_IRS_DATA_COLLECTORS_PERFORMANCE_REPORT_SLICE, fetchPlansParams).then(
+          result => {
+            fetchDataCollectors(result);
+          }
+        );
+      }
+      if (planId && jurisdictionId && dataCollector && !sopData.length) {
+        const fetchPlansParams = supersetFilters(params, dataCollector);
+        await service(SUPERSET_IRS_SOP_PERFORMANCE_REPORT_SLICE, fetchPlansParams).then(result => {
+          fetchSOPs(result);
+        });
+      }
+      if (planId && jurisdictionId && dataCollector && sop && !tableData.length) {
+        const fetchPlansParams = supersetFilters(params, sop);
+        await service(SUPERSET_IRS_SOP_BY_DATE_PERFORMANCE_REPORT_SLICE, fetchPlansParams).then(
+          result => {
+            fetchSopByDate(result);
+          }
+        );
+      }
+    } catch (e) {
+      // todo - handle error https://github.com/onaio/reveal-frontend/issues/300
```

Comment:
looks like this issue is already closed, maybe we can address the error here another way.